# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/ps_split_composite.glsl
# Commit: 15da94c6a7b1
# Full Hash: 15da94c6a7b19f971d3dffaa4cab8edb8e41cb62
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r 72ddac486037 -r 15da94c6a7b1 gfx/wr/webrender/res/ps_split_composite.glsl
--- a/gfx/wr/webrender/res/ps_split_composite.glsl	Tue Dec 15 01:46:51 2020 +0200
+++ b/gfx/wr/webrender/res/ps_split_composite.glsl	Mon Dec 14 21:48:41 2020 +0000
@@ -2,12 +2,14 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#define WR_FEATURE_TEXTURE_2D
+
 #include shared,prim_shared
 
 // interpolated UV coordinates to sample.
 varying vec2 vUv;
-// X = layer index to sample, Y = flag to allow perspective interpolation of UV.
-flat varying vec2 vLayerAndPerspective;
+// flag to allow perspective interpolation of UV.
+flat varying float vPerspective;
 flat varying vec4 vUvSampleBounds;
 
 #ifdef WR_VERTEX_SHADER
@@ -104,15 +106,15 @@
     float perspective_interpolate = float(ph.user_data.y);
 
     vUv = uv / texture_size * mix(gl_Position.w, 1.0, perspective_interpolate);
-    vLayerAndPerspective = vec2(res.layer, perspective_interpolate);
+    vPerspective = perspective_interpolate;
 }
 #endif
 
 #ifdef WR_FRAGMENT_SHADER
 void main(void) {
     float alpha = do_clip();
-    float perspective_divisor = mix(gl_FragCoord.w, 1.0, vLayerAndPerspective.y);
+    float perspective_divisor = mix(gl_FragCoord.w, 1.0, vPerspective);
     vec2 uv = clamp(vUv * perspective_divisor, vUvSampleBounds.xy, vUvSampleBounds.zw);
-    write_output(alpha * textureLod(sColor0, vec3(uv, vLayerAndPerspective.x), 0.0));
+    write_output(alpha * texture(sColor0, uv));
 }
 #endif